react/week4/my-app/src/App.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import './App.css';
+import context from "./context";
+import InputField from './components/InputField'
+import FetchUsers from './components/FetchUsers'
+import ShowData from './components/ShowData'
+
+const ContextProvider = context.Provider;
+
+class App extends React.Component {
+    state ={
+    users: [],
+    inputValue:'',
+    isLoading: true,
+    }
+
+    handelUserSearch =(e)=>{
+      const inputValue = this.state.inputValue;
+        if(inputValue){
+          FetchUsers(inputValue) 
+          .then(body=>{  
+            console.log(body)    
+            this.setState({       
+              users: body.items,      
+              isLoading: false     
+            })  
+          })
+          
+        }
+        
+        this.setState({  inputValue: e.target.value})
+      }
+    
+
+  render(){
+    const contextValue = {
+      users:this.state.users,
+      inputValue:this.state.inputValue,
+      isLoading: this.state.isLoading,
+      handelUserSearch: this.handelUserSearch,
+      };
+
+  return (
+      <ContextProvider value={contextValue}>
+          <div className="App">
+                  <h1>Github users search App.</h1>
+                  <InputField /> 
```

Comment:
I like that thought of making this component, however, I think leads to unnecessary indirection. Since we only use one value in `InputField`, handelUserSearch, I think it would be simpler for others to read, if you just created an input here like this:  
```suggestion
                  <input type='text' id=" " placeholder='search for user' onChange={this.handelUserSearch} />
```

Reply:
that right, but I do it for at I understande the procedure more ...